packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -186,24 +187,31 @@ export class CdkToolkit {
     }
 
     const stacks = stackCollection.stackArtifacts;
-    const assetBuildTime = options.assetBuildTime ?? AssetBuildTime.ALL_BEFORE_DEPLOY;
+    const assetBuildTime = options.assetBuildTime ?? AssetBuildTime.ALL_BEFORE_DEPLOY; // TODO: deal with this
```

Comment:
```suggestion
```

We shouldn't need to care about `assetBuildTime` in this context anymore

Reply:
We do need it because we'll derive `prebuildAssets: true|false` from this.